protokube/cmd/protokube/main.go

```diff
@@ -194,7 +194,14 @@ func run() error {
 		if clusterID == "" {
 			clusterID = osVolumes.ClusterID()
 		}
-
+	} else if cloud == "alicloud" {
+		glog.Info("Initializing AliCloud volumes")
+		aliVolumes, err := protokube.NewALIVolumes()
+		if err != nil {
+			glog.Errorf("Error initializing Aliyun: %q", err)
+			os.Exit(1)
```

Comment:
You sure exiting entirely is desired?

Reply:
I think it's what the other providers do - failure to initialize the cloud if we've been told to do it is pretty catastrophic.

That said, yes, we're probably overdue for a refactor of this function to return error!